isla-lib/src/ir/ssa.rs

```diff
@@ -157,13 +157,15 @@ pub enum BlockLoc {
     // Field locations contain the previous name so that we can update one field at a time
     Field(Box<BlockLoc>, SSAName, SSAName),
     Addr(Box<BlockLoc>),
+    // Index is not assignable and don't need SSA Name, so the ssa number is always -1.
+    Index(Box<BlockLoc>, SSAName),
```

Comment:
Shouldn't this just use a `u32` like `Loc`?

Reply:
Yes, it's essentially a wrapper of `u32`. I use SSAName for consistency in style.

Reply:
It should just be `u32`. It should only be a `SSAName` if it was originally a `Name` in the non-SSA IR.

`Name::from_u32(n)` could maybe have a more descriptive name - it is just creating a `Name` for the number, it's creating a name that indexes the `n`th element of the symbol table which is not what we want here.